src/Core/System/Snippet/DataTransfer/Metadata/MetadataCollection.php

```diff
@@ -0,0 +1,76 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\System\Snippet\DataTransfer\Metadata;
+
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Struct\Collection;
+
+/**
+ * @internal
+ *
+ * @extends Collection<MetadataEntry>
+ */
+#[Package('discovery')]
+class MetadataCollection extends Collection
+{
+    /**
+     * @param list<MetadataEntry> $elements
+     */
+    public function __construct(
+        array $elements = [],
+    ) {
+        $elements = array_column($elements, null, 'locale');
+
+        parent::__construct($elements);
+    }
+
+    /**
+     * @description Adds the given MetadataEntry if no entry exists for the same locale,
+     * or if the given entry has a different timestamp than the existing one.
+     */
+    public function addIfRequired(MetadataEntry $remoteEntry): void
+    {
+        $this->validateType($remoteEntry);
+
+        $localEntry = $this->get($remoteEntry->locale);
+
+        if ($localEntry && $this->isUpToDate($localEntry, $remoteEntry)) {
+            return;
+        }
+
+        $remoteEntry->markForUpdate();
+        $this->elements[$remoteEntry->locale] = $remoteEntry;
+    }
+
+    public function toJson(): string
```

Comment:
why implementing such method, if there is already json_serialize, which could be extended?

Reply:
because the json_serialize method has a different return type - i want to have the elements as a json string to directly write the content on the file system